### ISSUE TITLE
Script error occurs in basic authentication mode

### ISSUE THREAD

**Issue by daisuke-yoshimoto** [Activiti/activiti-modeling-app#30]
**Type of issue:**  (check with "[x]")

> - [ ] New feature request
> - [x] Bug  
> - [ ] Support request
> - [ ] Documentation

**Current behaviour:**

```
core.js:12632 ERROR TypeError: Cannot read property 'callCustomApi' of undefined
    at RequestApiHelper.push../dist/ama-sdk/fesm5/ama-sdk.js.RequestApiHelper.request (ama-sdk.js:2565)
    at RequestApiHelper.push../dist/ama-sdk/fesm5/ama-sdk.js.RequestApiHelper.get (ama-sdk.js:2500)
    at ACMProjectApi.push../dist/ama-sdk/fesm5/ama-sdk.js.ACMProjectApi.getAll (ama-sdk.js:2671)
    at DashboardService.push../src/app/dashboard/services/dashboard.service.ts.DashboardService.fetchProjects (dashboard.service.ts:27)
    at ProjectsEffects.push../src/app/dashboard/store/effects/projects.effects.ts.ProjectsEffects.getProjectsAttempt (projects.effects.ts:149)
    at SwitchMapSubscriber.project (projects.effects.ts:106)
    at SwitchMapSubscriber.push../node_modules/rxjs/_esm5/internal/operators/switchMap.js.SwitchMapSubscriber._next (switchMap.js:35)
    at SwitchMapSubscriber.push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber.next (Subscriber.js:54)
    at FilterSubscriber.push../node_modules/rxjs/_esm5/internal/operators/filter.js.FilterSubscriber._next (filter.js:38)
    at FilterSubscriber.push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber.next (Subscriber.js:54)
defaultErrorLogger @ core.js:12632
push../node_modules/@angular/core/fesm5/core.js.ErrorHandler.handleError @ core.js:12680
reportErrorThrown @ effects.js:211
verifyOutput @ effects.js:206
(anonymous) @ effects.js:273
push../node_modules/rxjs/_esm5/internal/operators/map.js.MapSubscriber._next @ map.js:35
push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber.next @ Subscriber.js:54
push../node_modules/rxjs/_esm5/internal/operators/exhaustMap.js.ExhaustMapSubscriber.notifyNext @ exhaustMap.js:67
push../node_modules/rxjs/_esm5/internal/InnerSubscriber.js.InnerSubscriber._next @ InnerSubscriber.js:15
push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber.next @ Subscriber.js:54
push../node_modules/rxjs/_esm5/internal/operators/mergeMap.js.MergeMapSubscriber.notifyNext @ mergeMap.js:84
push../node_modules/rxjs/_esm5/internal/InnerSubscriber.js.InnerSubscriber._next @ InnerSubscriber.js:15
push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber.next @ Subscriber.js:54
push../node_modules/rxjs/_esm5/internal/operators/map.js.MapSubscriber._next @ map.js:41
push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber.next @ Subscriber.js:54
push../node_modules/rxjs/_esm5/internal/operators/materialize.js.MaterializeSubscriber._error @ materialize.js:28
push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber.error @ Subscriber.js:60
push../node_modules/rxjs/_esm5/internal/operators/switchMap.js.SwitchMapSubscriber._next @ switchMap.js:38
push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber.next @ Subscriber.js:54
push../node_modules/rxjs/_esm5/internal/operators/filter.js.FilterSubscriber._next @ filter.js:38
push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber.next @ Subscriber.js:54
push../node_modules/rxjs/_esm5/internal/Subject.js.Subject.next @ Subject.js:47
(anonymous) @ store-devtools.js:657
push../node_modules/rxjs/_esm5/internal/Subscriber.js.SafeSubscriber.__tryOrUnsub @ Subscriber.js:196
push../node_modules/rxjs/_esm5/internal/Subscriber.js.SafeSubscriber.next @ Subscriber.js:134
push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber._next @ Subscriber.js:77
push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber.next @ Subscriber.js:54
push../node_modules/rxjs/_esm5/internal/operators/scan.js.ScanSubscriber._tryNext @ scan.js:67
push../node_modules/rxjs/_esm5/internal/operators/scan.js.ScanSubscriber._next @ scan.js:54
push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber.next @ Subscriber.js:54
push../node_modules/rxjs/_esm5/internal/operators/withLatestFrom.js.WithLatestFromSubscriber._next @ withLatestFrom.js:66
push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber.next @ Subscriber.js:54
push../node_modules/rxjs/_esm5/internal/Notification.js.Notification.observe @ Notification.js:15
push../node_modules/rxjs/_esm5/internal/operators/observeOn.js.ObserveOnSubscriber.dispatch @ observeOn.js:40
push../node_modules/rxjs/_esm5/internal/scheduler/AsyncAction.js.AsyncAction._execute @ AsyncAction.js:63
push../node_modules/rxjs/_esm5/internal/scheduler/QueueAction.js.QueueAction.execute @ QueueAction.js:27
push../node_modules/rxjs/_esm5/internal/scheduler/AsyncScheduler.js.AsyncScheduler.flush @ AsyncScheduler.js:43
push../node_modules/rxjs/_esm5/internal/scheduler/QueueAction.js.QueueAction.schedule @ QueueAction.js:21
push../node_modules/rxjs/_esm5/internal/Scheduler.js.Scheduler.schedule @ Scheduler.js:13
push../node_modules/rxjs/_esm5/internal/scheduler/AsyncScheduler.js.AsyncScheduler.schedule @ AsyncScheduler.js:31
push../node_modules/rxjs/_esm5/internal/operators/observeOn.js.ObserveOnSubscriber.scheduleMessage @ observeOn.js:45
push../node_modules/rxjs/_esm5/internal/operators/observeOn.js.ObserveOnSubscriber._next @ observeOn.js:48
push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber.next @ Subscriber.js:54
push../node_modules/rxjs/_esm5/internal/operators/mergeMap.js.MergeMapSubscriber.notifyNext @ mergeMap.js:84
push../node_modules/rxjs/_esm5/internal/InnerSubscriber.js.InnerSubscriber._next @ InnerSubscriber.js:15
push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber.next @ Subscriber.js:54
push../node_modules/rxjs/_esm5/internal/operators/map.js.MapSubscriber._next @ map.js:41
push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber.next @ Subscriber.js:54
push../node_modules/rxjs/_esm5/internal/operators/mergeMap.js.MergeMapSubscriber.notifyNext @ mergeMap.js:84
push../node_modules/rxjs/_esm5/internal/InnerSubscriber.js.InnerSubscriber._next @ InnerSubscriber.js:15
push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber.next @ Subscriber.js:54
push../node_modules/rxjs/_esm5/internal/operators/skip.js.SkipSubscriber._next @ skip.js:26
push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber.next @ Subscriber.js:54
push../node_modules/rxjs/_esm5/internal/Subject.js.Subject.next @ Subject.js:47
push../node_modules/rxjs/_esm5/internal/BehaviorSubject.js.BehaviorSubject.next @ BehaviorSubject.js:38
push../node_modules/@ngrx/store/fesm5/store.js.ActionsSubject.next @ store.js:33
push../node_modules/@ngrx/store/fesm5/store.js.Store.dispatch @ store.js:416
push../src/app/dashboard/router/guards/projects-loader.guard.ts.ProjectsLoaderGuard.canActivate @ projects-loader.guard.ts:33
(anonymous) @ router.js:3004
push../node_modules/rxjs/_esm5/internal/operators/map.js.MapSubscriber._next @ map.js:35
push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber.next @ Subscriber.js:54
(anonymous) @ subscribeToArray.js:5
push../node_modules/rxjs/_esm5/internal/Observable.js.Observable._trySubscribe @ Observable.js:43
push../node_modules/rxjs/_esm5/internal/Observable.js.Observable.subscribe @ Observable.js:29
push../node_modules/rxjs/_esm5/internal/operators/map.js.MapOperator.call @ map.js:18
push../node_modules/rxjs/_esm5/internal/Observable.js.Observable.subscribe @ Observable.js:24
push../node_modules/rxjs/_esm5/internal/operators/mergeMap.js.MergeMapOperator.call @ mergeMap.js:29
push../node_modules/rxjs/_esm5/internal/Observable.js.Observable.subscribe @ Observable.js:24
push../node_modules/rxjs/_esm5/internal/operators/every.js.EveryOperator.call @ every.js:14
push../node_modules/rxjs/_esm5/internal/Observable.js.Observable.subscribe @ Observable.js:24
(anonymous) @ subscribeTo.js:21
subscribeToResult @ subscribeToResult.js:11
push../node_modules/rxjs/_esm5/internal/operators/mergeMap.js.MergeMapSubscriber._innerSub @ mergeMap.js:74
push../node_modules/rxjs/_esm5/internal/operators/mergeMap.js.MergeMapSubscriber._tryNext @ mergeMap.js:68
push../node_modules/rxjs/_esm5/internal/operators/mergeMap.js.MergeMapSubscriber._next @ mergeMap.js:51
push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber.next @ Subscriber.js:54
(anonymous) @ subscribeToArray.js:5
push../node_modules/rxjs/_esm5/internal/Observable.js.Observable._trySubscribe @ Observable.js:43
push../node_modules/rxjs/_esm5/internal/Observable.js.Observable.subscribe @ Observable.js:29
push../node_modules/rxjs/_esm5/internal/operators/mergeMap.js.MergeMapOperator.call @ mergeMap.js:29
push../node_modules/rxjs/_esm5/internal/Observable.js.Observable.subscribe @ Observable.js:24
push../node_modules/rxjs/_esm5/internal/operators/every.js.EveryOperator.call @ every.js:14
push../node_modules/rxjs/_esm5/internal/Observable.js.Observable.subscribe @ Observable.js:24
(anonymous) @ subscribeTo.js:21
subscribeToResult @ subscribeToResult.js:11
push../node_modules/rxjs/_esm5/internal/operators/mergeMap.js.MergeMapSubscriber._innerSub @ mergeMap.js:74
push../node_modules/rxjs/_esm5/internal/operators/mergeMap.js.MergeMapSubscriber._tryNext @ mergeMap.js:68
push../node_modules/rxjs/_esm5/internal/operators/mergeMap.js.MergeMapSubscriber._next @ mergeMap.js:51
push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber.next @ Subscriber.js:54
(anonymous) @ subscribeToArray.js:5
push../node_modules/rxjs/_esm5/internal/Observable.js.Observable._trySubscribe @ Observable.js:43
push../node_modules/rxjs/_esm5/internal/Observable.js.Observable.subscribe @ Observable.js:29
push../node_modules/rxjs/_esm5/internal/operators/mergeMap.js.MergeMapOperator.call @ mergeMap.js:29
push../node_modules/rxjs/_esm5/internal/Observable.js.Observable.subscribe @ Observable.js:24
push../node_modules/rxjs/_esm5/internal/operators/every.js.EveryOperator.call @ every.js:14
push../node_modules/rxjs/_esm5/internal/Observable.js.Observable.subscribe @ Observable.js:24
(anonymous) @ subscribeTo.js:21
subscribeToResult @ subscribeToResult.js:11
push../node_modules/rxjs/_esm5/internal/operators/mergeMap.js.MergeMapSubscriber._innerSub @ mergeMap.js:74
push../node_modules/rxjs/_esm5/internal/operators/mergeMap.js.MergeMapSubscriber._tryNext @ mergeMap.js:68
push../node_modules/rxjs/_esm5/internal/operators/mergeMap.js.MergeMapSubscriber._next @ mergeMap.js:51
push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber.next @ Subscriber.js:54
push../node_modules/rxjs/_esm5/internal/operators/every.js.EverySubscriber.notifyComplete @ every.js:30
push../node_modules/rxjs/_esm5/internal/operators/every.js.EverySubscriber._complete @ every.js:47
push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber.complete @ Subscriber.js:66
push../node_modules/rxjs/_esm5/internal/operators/mergeMap.js.MergeMapSubscriber._complete @ mergeMap.js:79
push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber.complete @ Subscriber.js:66
(anonymous) @ subscribeToArray.js:8
push../node_modules/rxjs/_esm5/internal/Observable.js.Observable._trySubscribe @ Observable.js:43
push../node_modules/rxjs/_esm5/internal/Observable.js.Observable.subscribe @ Observable.js:29
push../node_modules/rxjs/_esm5/internal/operators/mergeMap.js.MergeMapOperator.call @ mergeMap.js:29
push../node_modules/rxjs/_esm5/internal/Observable.js.Observable.subscribe @ Observable.js:24
push../node_modules/rxjs/_esm5/internal/operators/every.js.EveryOperator.call @ every.js:14
push../node_modules/rxjs/_esm5/internal/Observable.js.Observable.subscribe @ Observable.js:24
push../node_modules/rxjs/_esm5/internal/operators/mergeMap.js.MergeMapOperator.call @ mergeMap.js:29
push../node_modules/rxjs/_esm5/internal/Observable.js.Observable.subscribe @ Observable.js:24
push../node_modules/rxjs/_esm5/internal/operators/map.js.MapOperator.call @ map.js:18
push../node_modules/rxjs/_esm5/internal/Observable.js.Observable.subscribe @ Observable.js:24
(anonymous) @ subscribeTo.js:21
subscribeToResult @ subscribeToResult.js:11
push../node_modules/rxjs/_esm5/internal/operators/mergeMap.js.MergeMapSubscriber._innerSub @ mergeMap.js:74
push../node_modules/rxjs/_esm5/internal/operators/mergeMap.js.MergeMapSubscriber._tryNext @ mergeMap.js:68
push../node_modules/rxjs/_esm5/internal/operators/mergeMap.js.MergeMapSubscriber._next @ mergeMap.js:51
push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber.next @ Subscriber.js:54
push../node_modules/rxjs/_esm5/internal/operators/map.js.MapSubscriber._next @ map.js:41
push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber.next @ Subscriber.js:54
push../node_modules/rxjs/_esm5/internal/operators/tap.js.TapSubscriber._next @ tap.js:51
push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber.next @ Subscriber.js:54
push../node_modules/rxjs/_esm5/internal/operators/switchMap.js.SwitchMapSubscriber.notifyNext @ switchMap.js:72
push../node_modules/rxjs/_esm5/internal/InnerSubscriber.js.InnerSubscriber._next @ InnerSubscriber.js:15
push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber.next @ Subscriber.js:54
push../node_modules/rxjs/_esm5/internal/operators/map.js.MapSubscriber._next @ map.js:41
push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber.next @ Subscriber.js:54
(anonymous) @ scalar.js:5
push../node_modules/rxjs/_esm5/internal/Observable.js.Observable._trySubscribe @ Observable.js:43
push../node_modules/rxjs/_esm5/internal/Observable.js.Observable.subscribe @ Observable.js:29
push../node_modules/rxjs/_esm5/internal/operators/map.js.MapOperator.call @ map.js:18
push../node_modules/rxjs/_esm5/internal/Observable.js.Observable.subscribe @ Observable.js:24
(anonymous) @ subscribeTo.js:21
subscribeToResult @ subscribeToResult.js:11
push../node_modules/rxjs/_esm5/internal/operators/switchMap.js.SwitchMapSubscriber._innerSub @ switchMap.js:51
push../node_modules/rxjs/_esm5/internal/operators/switchMap.js.SwitchMapSubscriber._next @ switchMap.js:41
push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber.next @ Subscriber.js:54
push../node_modules/rxjs/_esm5/internal/operators/switchMap.js.SwitchMapSubscriber.notifyNext @ switchMap.js:72
push../node_modules/rxjs/_esm5/internal/InnerSubscriber.js.InnerSubscriber._next @ InnerSubscriber.js:15
push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber.next @ Subscriber.js:54
push../node_modules/rxjs/_esm5/internal/operators/tap.js.TapSubscriber._next @ tap.js:51
push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber.next @ Subscriber.js:54
push../node_modules/rxjs/_esm5/internal/operators/mergeMap.js.MergeMapSubscriber.notifyNext @ mergeMap.js:84
push../node_modules/rxjs/_esm5/internal/InnerSubscriber.js.InnerSubscriber._next @ InnerSubscriber.js:15
push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber.next @ Subscriber.js:54
push../node_modules/rxjs/_esm5/internal/operators/map.js.MapSubscriber._next @ map.js:41
push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber.next @ Subscriber.js:54
(anonymous) @ scalar.js:5
push../node_modules/rxjs/_esm5/internal/Observable.js.Observable._trySubscribe @ Observable.js:43
push../node_modules/rxjs/_esm5/internal/Observable.js.Observable.subscribe @ Observable.js:29
push../node_modules/rxjs/_esm5/internal/operators/map.js.MapOperator.call @ map.js:18
push../node_modules/rxjs/_esm5/internal/Observable.js.Observable.subscribe @ Observable.js:24
(anonymous) @ subscribeTo.js:21
subscribeToResult @ subscribeToResult.js:11
push../node_modules/rxjs/_esm5/internal/operators/mergeMap.js.MergeMapSubscriber._innerSub @ mergeMap.js:74
push../node_modules/rxjs/_esm5/internal/operators/mergeMap.js.MergeMapSubscriber._tryNext @ mergeMap.js:68
push../node_modules/rxjs/_esm5/internal/operators/mergeMap.js.MergeMapSubscriber._next @ mergeMap.js:51
push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber.next @ Subscriber.js:54
push../node_modules/rxjs/_esm5/internal/operators/switchMap.js.SwitchMapSubscriber.notifyNext @ switchMap.js:72
push../node_modules/rxjs/_esm5/internal/InnerSubscriber.js.InnerSubscriber._next @ InnerSubscriber.js:15
push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber.next @ Subscriber.js:54
push../node_modules/rxjs/_esm5/internal/operators/map.js.MapSubscriber._next @ map.js:41
push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber.next @ Subscriber.js:54
push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber._next @ Subscriber.js:77
push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber.next @ Subscriber.js:54
push../node_modules/rxjs/_esm5/internal/operators/map.js.MapSubscriber._next @ map.js:41
push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber.next @ Subscriber.js:54
push../node_modules/rxjs/_esm5/internal/operators/map.js.MapSubscriber._next @ map.js:41
push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber.next @ Subscriber.js:54
push../node_modules/rxjs/_esm5/internal/operators/map.js.MapSubscriber._next @ map.js:41
push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber.next @ Subscriber.js:54
push../node_modules/rxjs/_esm5/internal/operators/tap.js.TapSubscriber._next @ tap.js:51
push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber.next @ Subscriber.js:54
push../node_modules/rxjs/_esm5/internal/operators/takeLast.js.TakeLastSubscriber._complete @ takeLast.js:57
push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber.complete @ Subscriber.js:66
push../node_modules/rxjs/_esm5/internal/operators/mergeMap.js.MergeMapSubscriber.notifyComplete @ mergeMap.js:94
push../node_modules/rxjs/_esm5/internal/InnerSubscriber.js.InnerSubscriber._complete @ InnerSubscriber.js:22
push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber.complete @ Subscriber.js:66
push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber._complete @ Subscriber.js:84
push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber.complete @ Subscriber.js:66
push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber._complete @ Subscriber.js:84
push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber.complete @ Subscriber.js:66
push../node_modules/rxjs/_esm5/internal/operators/tap.js.TapSubscriber._complete @ tap.js:71
push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber.complete @ Subscriber.js:66
push../node_modules/rxjs/_esm5/internal/operators/take.js.TakeSubscriber._next @ take.js:42
push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber.next @ Subscriber.js:54
push../node_modules/rxjs/_esm5/internal/operators/filter.js.FilterSubscriber._next @ filter.js:38
push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber.next @ Subscriber.js:54
push../node_modules/rxjs/_esm5/internal/operators/mergeMap.js.MergeMapSubscriber.notifyNext @ mergeMap.js:84
push../node_modules/rxjs/_esm5/internal/InnerSubscriber.js.InnerSubscriber._next @ InnerSubscriber.js:15
push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber.next @ Subscriber.js:54
push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber._next @ Subscriber.js:77
push../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber.next @ Subscriber.js:54
push../node_modules/rxjs/_esm5/internal/operators/mergeMap.js.MergeMapSubscriber.notifyNext @ mergeMap.js:84
Promise.then (async)
scheduleMicroTask @ zone.js:578
push../node_modules/zone.js/dist/zone.js.ZoneDelegate.scheduleTask @ zone.js:410
onScheduleTask @ zone.js:297
push../node_modules/zone.js/dist/zone.js.ZoneDelegate.scheduleTask @ zone.js:401
push../node_modules/zone.js/dist/zone.js.Zone.scheduleTask @ zone.js:232
push../node_modules/zone.js/dist/zone.js.Zone.scheduleMicroTask @ zone.js:252
scheduleResolveOrReject @ zone.js:862
resolvePromise @ zone.js:808
(anonymous) @ zone.js:724
webpackJsonpCallback @ bootstrap:25
(anonymous) @ dashboard-dashboard-module.js:1
```

* https://github.com/Activiti/activiti-modeling-app/blob/master/projects/ama-sdk/src/lib/api-implementations/acm-api/request-api.helper.ts#L56
* https://github.com/Activiti/activiti-modeling-app/blob/master/projects/ama-sdk/src/lib/api-implementations/acm-api/request-api.helper.ts#L107

**Steps to reproduce the issue:**
* Change authType setting from OAUTH to BASIC in [app.config.json](https://github.com/Activiti/activiti-modeling-app/blob/master/src/app.config.json#L5).
* Enter userId/password and press the Logic button
* Script error occurs 

**Component name and version:** 

**Browser and version:** 
Chrome 71
 
**Node version (for build issues):**  
v10.12.0

**Comment by daisuke-yoshimoto** [comment]
@salaboy @mteodori 

It seems to work even in the case of basic authentication by adding a branch when oauth2Auth does not exist as follows.

https://github.com/Activiti/activiti-modeling-app/blob/master/projects/ama-sdk/src/lib/api-implementations/acm-api/request-api.helper.ts#L56

### Before

```
private get api() {
        return this.alfrescoApiService.getInstance().oauth2Auth;
}
```

### After

```
private get api() {
    var alfrescoApiService = this.alfrescoApiService.getInstance();
    if (alfrescoApiService.oauth2Auth) {
        return alfrescoApiService.oauth2Auth;            
    } else {
        return alfrescoApiService.ecmClient;
    }
}
```

**Comment by salaboy** [comment]
@daisuke-yoshimoto so just to clarify... with:
```
private get api() {
    var alfrescoApiService = this.alfrescoApiService.getInstance();
    if (alfrescoApiService.oauth2Auth) {
        return alfrescoApiService.oauth2Auth;            
    } else {
        return alfrescoApiService.ecmClient;
    }
}
```

Does it work? 

@popovicsandras can we add this code to support basicAuth? It looks lik ecmClient is providing that functionality??

**Comment by popovicsandras** [comment]
Sorry, can someone give me a little bit of more context?
We can add the code above for sure, but I don't see how the content service (for which ecmClient exists) fits into the picture of a process modeling application.
What do we want to use the `BASIC` authentication for.
Until now, my understanding was that, this application is supposed to work with an ACM-like service.
What is changed about the requirements?

**Comment by popovicsandras** [comment]
Ok, let's discuss it on Monday, this one seems like a new request for me, which I'd like to know more about. 

I don't think just simply enabling one of ecmClient or bpmClient could solve this problem, as of @daisuke-yoshimoto already raised some issues about it. This is because of that, we are talking about a more complex scenario, where depending on the preferred strategy the following components may need to behave differently:

- authentication
- authorisation
- request sending
- router guards
- user menu

**Comment by epiard13** [comment]
Hi there,
I'm trying to use this app, i got it running and everything, but for the life of me i'm not able to login.
- Is there a default login?
- Is this Angular app suppose to work by itself, or does it need a backend-end app? if so which one?
- I've search the whole internet on how to make the login work, i've not been successful, i've even tried the docker examples, but would either run into issues with keycloak or other stuff.

Basically this is what i'm trying to accomplish : 
- Create a diagram, export it and import the xml process into a Runtime app (which i've already setup)

Please advise thx.

**Comment by salaboy** [comment]
@epiard13 can you please let us know how are you running the application? and which version are you using?

**Comment by fcorti** [comment]
Closing this issue, because pretty old.
Feel free to reopen it (or open a new one) if needed.

Cheers.
